odoo/orm/fields_textual.py

```diff
@@ -291,25 +386,19 @@ def write(self, records, value):
                 new_store_translations['en_US'] = cache_value
                 new_store_translations.pop('_en_US', None)
             new_translations_list.append(new_store_translations)
-        # Maybe we can use Cache.update(records.with_context(cache_update_raw=True), self, new_translations_list, dirty=True)
-        cache.update_raw(records, self, new_translations_list, dirty=True)
+        for record, new_translation in zip(records.with_context(prefetch_langs=True), new_translations_list, strict=True):
+            self._cache_update(record, new_translation, dirty=True)
 
     def to_sql(self, model: BaseModel, alias: str, flush: bool = True) -> SQL:
         sql_field = super().to_sql(model, alias, flush)
-        if self.translate:
+        if self.translate and not model.env.context.get('prefetch_langs'):
```

Comment:
I guess this will case a bug.
Semantically, `prefetch_langs` should only be used for prefetch and affect `SELECT`
But `to_sql` will also be used for `condition_to_sql` to generate `WHERE` condition
e.g. `Model.with_context(prefetch_langs=True).search(['name', 'ilike', 'xxx'])` will have condition `table.name ilike '%xxx%'` for jsonb

Merging `to_sql` and `to_raw_sql` seems has less apis, but it becomes more tricky since the result is context dependent now.

Reply:
The result of most functions of translations is already context-dependent. I will force prefetch_langs=False when translating conditions.

Reply:
Another use case is `_field_to_sql` which is used to generate SQL queries in account
Technically, we always need `model.with_context(prefetch_langs=False)._field_to_sql(` for translated fields

I think usually, we prefer more apis than implicit arguments. But if we agree with using the context here. I think it would be better to use an extra technical context and define it only in the `fetch`